tiatoolbox/utils/misc.py

```diff
@@ -576,6 +607,58 @@ def unzip_data(zip_path, save_path, del_zip=True):
         os.remove(zip_path)
 
 
+def walk_list_dict(in_list_dict):
+    """Recursive walk and jsonify in place.
+
+    Args:
+        in_list_dict (list or dict):  input list or a dictionary.
+
+    Returns:
+        list or dict
+
+    """
+    if isinstance(in_list_dict, dict):
+        walk_dict(in_list_dict)
+    elif isinstance(in_list_dict, list):
+        walk_list(in_list_dict)
+    elif isinstance(in_list_dict, np.ndarray):
+        in_list_dict = in_list_dict.tolist()
+        walk_list(in_list_dict)
+    elif isinstance(in_list_dict, np.generic):
+        in_list_dict = in_list_dict.item()
+    elif in_list_dict is not None and not isinstance(
+        in_list_dict, (int, float, str, bool)
+    ):
+        raise ValueError(
+            f"Value type `{type(in_list_dict)}` `{in_list_dict}` is not jsonified."
+        )
+    return in_list_dict
+
+
+def walk_list(lst):
+    """Recursive walk and jsonify a list in place.
+
+    Args:
+        lst (list):  input list.
+
+    """
+    for i, v in enumerate(lst):
+        lst[i] = walk_list_dict(v)
```

Comment:
I think this should be turned into a private function, we dont want to expose it to other file.